redash/query_runner/prometheus.py

```diff
@@ -40,52 +89,66 @@ def get_schema(self, get_stats=False):
+            query=http_requests_total
+
+        example: range query
+            query=http_requests_total&start=2018-01-20T00:00:00.000Z&end=2018-01-25T00:00:00.000Z&step=60s
+
+        example: until now range query
+            query=http_requests_total&start=2018-01-20T00:00:00.000Z&step=60s
+            query=http_requests_total&start=2018-01-20T00:00:00.000Z&end=now&step=60s
+        """
+
         base_url = self.configuration["url"]
 
         try:
             error = None
             query = query.strip()
+            # for backward compatibility
+            query = 'query={}'.format(query) if not query.startswith('query=') else query
 
-            local_query = '/api/v1/query'
-            url = base_url + local_query
-            payload = {'query': query}
-            response = requests.get(url, params=payload)
+            payload = parse_qs(query)
+            query_type = 'query_range' if 'step' in payload.keys() else 'query'
+
+            # for the range of until now
+            if query_type == 'query_range' and ('end' not in payload.keys() or 'now' in payload['end']):
+                date_now = datetime.now()
+                payload.update({"end": [date_now.isoformat("T") + "Z"]})
+
+            api_endpoint = base_url + '/api/v1/{}'.format(query_type)
+
+            response = requests.get(api_endpoint, params=payload)
             response.raise_for_status()
-            raw_data = response.json()['data']['result']
-            columns = [
-                {
-                    'friendly_name': 'timestamp',
-                    'type': TYPE_DATETIME,
-                    'name': 'timestamp'
-                },
-                {
-                    'friendly_name': 'value',
-                    'type': TYPE_STRING,
-                    'name': 'value'
-                },
-            ]
-            columns_name = raw_data[0]['metric'].keys()
-            for column_name in columns_name:
-                columns.append({
-                    'friendly_name': column_name,
+
+            metrics = response.json()['data']['result']
+
+            if len(metrics) == 0:
+                return None, 'query result is empty.'
+
+            metric_labels = metrics[0]['metric'].keys()
+
+            for label_name in metric_labels:
+                self.columns.append({
+                    'friendly_name': label_name,
                     'type': TYPE_STRING,
-                    'name': column_name
+                    'name': label_name
                 })
-            rows = []
-            for row in raw_data:
-                h = {}
-                for r in row['metric']:
-                    h[r] = row['metric'][r]
-                    h['value'] = row['value'][1]
-                    h['timestamp'] = datetime.datetime.fromtimestamp(row['value'][0])
-                rows.append(h)
 
+            rows = get_range_rows(metrics) if query_type == 'query_range' else get_instant_rows(metrics)
```

Comment:
I think this will be easier to read as a regular `if`:

```python
if query_type == 'query_range' :
    rows = get_range_rows(metrics) 
else:
    rows = get_instant_rows(metrics)
```

(there is no need to predefine `rows` as `if` doesn't create a new scope in Python)

Reply:
fixed.